### PR TITLE
Fix user yield generated decimals

### PR DESCRIPTION
**Pull request by maikelordaz**
Related to git issue #164

### CODE REVIEW

File: contracts/facets/GettersFacet.sol [hunk 1]
```diff
@@ -647,10 +647,10 @@ contract GettersFacet is IGetters {
             ._yieldStorage()
             .yields[termId];
 
-        return
-            yield.withdrawnYield[user] +
-            totalYieldGenerated(termId) *
-            yieldDistributionRatio(termId, user);
+        uint yieldDistributed = (totalYieldGenerated(termId) *
```
Comment (MohammedTaka) [inline contracts/facets/GettersFacet.sol]: totalYieldGenerated is not correctly used. totalYieldGenerated includes current yield in zaynfi and withdrawn yield of everyone. This last bit should not included in the userYieldGenerated.

So for totalYieldGenerated add a new function that doesn't include withdrawnYield

File: contracts/facets/GettersFacet.sol [hunk 2]
```diff
@@ -647,10 +670,10 @@ contract GettersFacet is IGetters {
...
             .yields[termId];
 
-        return
-            yield.withdrawnYield[user] +
-            totalYieldGenerated(termId) *
-            yieldDistributionRatio(termId, user);
+        uint yieldDistributed = (currentYieldGenerated(termId) *
+            yieldDistributionRatio(termId, user)) / 10 ** 18;
+
+        return yield.withdrawnYield[user] + yieldDistributed;
```
Comment (elallamTaka) [inline contracts/facets/GettersFacet.sol]: Only return the `yieldDistributed`. Perhaps call the function `unwithdrawnUserYieldGenerated `
Append `yieldDistributed` to `getUserYieldSummary `.
The frontend will then only call `getUserYieldSummary `and display `withdrawnYield + yieldDistributed`

**Review by elallamTaka — CHANGES_REQUESTED** [review]

File: contracts/facets/GettersFacet.sol (hunk 2 above)
Comment (maikelordaz) [reply contracts/facets/GettersFacet.sol]: Applied on the latests commits

Comment (MohammedTaka) [reply contracts/facets/GettersFacet.sol]: Please move unwithddrawnUserYieldGenerated to LibyieldGeneration, you can also make currentYieldGenerated internal and move it, as it depends on it.

**Review by elallamTaka — APPROVED** [review]

**Review by MohammedTaka — APPROVED** [review]

**Pull request merged**